doc/ext-pake/proposed/spake2plus-proposal-draft.md

```diff
@@ -0,0 +1,66 @@
+Introduction
+------------
+
+This document proposes protocol flow and high-level API changes required in PSA Cryptography API 1.1 PAKE Extension, to support the SPAKE2+ protocol.
+
+SPAKE2+ Version
+---------------
+
+SPAKE2+, an Augmented PAKE Draft 02, 10 December 2020 is considered for proposal.
+
+Link : [https://tools.ietf.org/pdf/draft-bar-cfrg-spake2plus-02.pdf](https://tools.ietf.org/pdf/draft-bar-cfrg-spake2plus-02.pdf)
+
+Remarks
+
+*   SPAKE2+, an Augmented PAKE Draft 08, 5 May 2022 is the latest draft version. Link : [https://datatracker.ietf.org/doc/pdf/draft-bar-cfrg-spake2plus-08](https://datatracker.ietf.org/doc/pdf/draft-bar-cfrg-spake2plus-08)
+*   Shared Secret Key generation is not compatible between Draft 02 and 08.
+*   As most SPAKE2+ implementations e.g. Matter Specification Version 1.0 are based on Draft 02, this version is being considered for better interoperability.
+
+Expected PAKE API Flow :
+------------------------------------
+![](SPAKE2PLUS.svg)
+
+
+
+#### Out of band set up
+
+SPAKE2+ has an offline initialization step where both parties may share parties identities and additional data (the context). This step is considered out of scope but these shared information may be required for SPAKE2+ protocol execution.
+
+**Shared Information** : ProverID, VerifierID and Context.
+
+**Input methods** :
+
+*   Existing API's (when user is Prover)
+    *   `psa_pake_set_user()` to input ProverID
+    *   `psa_pake_set_peer()` to input VerifierID
+*   Proposed new API
+    *   `psa_pake_set_context()` to input context (additional data).
+
+#### Password, PBKDF Output and Verification value (W0, W1, L)
+
+SPAKE2+ requires PBKDF Output and Verification value (W0, W1, L) generated from password as below :
+
+ `w0s || w1s = PBKDF(len(pw) || pw || len(A) || A || len(B) || B)`  
+ `W0 = w0s mod p`  
+ `W1 = w1s mod p`  
+ `L = w1*P`
+
+Computing these values is considered Out of Scope for PAKE API as this a registration phase which could happen out of actual protocol flow.
+
+Since W0 & W1 and L serve as registartion records and verification value, these values can be treated as secrets.
```

Comment:
The standard allows/suggests several different formats, but here I think it would make sense to mandate the same format as for ECC private (W0 and W1) and public (L) keys.

Reply:
(For what it worths, I can't think of any use case where these values would need to be visible outside of the key store.)

Reply:
It is possible and I agree that it is how the registration flow should go. However, nothing should be sent in the clear. Whatever the API will be, we need to emphasise, that this is a setup to the protocol, that has to go through secure channels.

> eavesdropper who sees w0 and L is unable to spoof the Prover role

That is true, but they still can impersonate the Verifier, which is undesirable as well.

Reply:
Thanks for inputs. Let me check possible application use cases for secret usage in SPAKE2+ and get back on this.